evalml/automl/engine/sequential_engine.py

```diff
@@ -20,7 +21,10 @@ def evaluate_batch(self, pipelines):
         while self._should_continue_callback() and index < len(pipelines):
             pipeline = pipelines[index]
             self._pre_evaluation_callback(pipeline)
-            evaluation_result = EngineBase.train_and_score_pipeline(pipeline, self.automl, self.X_train, self.y_train)
+            if pipeline.model_family == ModelFamily.ENSEMBLE:
```

Comment:
Sure, that works for me.  Would have to see what that looks like with the call to super().  Do you mean putting this whole if/else branch in super and swapping the X/y?

Reply:
It's trickier than I thought since the logic depends on a single pipeline but the function takes in a list of pipelines. Best to do it later but it would be nice if we didn't have to replicate this along all engines.

Reply:
@chukarsten @freddyaboulton what if we did something like
```python
# in engine_base.py
# in engine class
def handle_train_and_score_pipeline(pipeline):
    if pipeline.model_family == ModelFamily.ENSEMBLE:
         evaluation_result = EngineBase.train_and_score_pipeline(pipeline, self.automl, self.X_ensemble, self.y_ensemble)
    else:
         evaluation_result = EngineBase.train_and_score_pipeline(pipeline, self.automl, self.X_train, self.y_train)
    return evaluation_result
```

Then in `SequentialEngine`, you can simply call `super().handle_train_and_score_pipeline(pipeline)`? 

This way, it'll be in the base Engine class, although we wouldn't be adding it to `EngineBase.evaluate_batch`, and it might just be an extra method call that we don't need.

Reply:
@bchen1116 I like that! I think we'll need to do re-use the logic for #1729 so it may be best to return the data to use (`X_ensemble` vs `X_train`) rather than the evaluation result.

I think it's best for me to handle this in the PR for #1729. I'm convinced this is scope creep now cause the best way to do this depends on requirements not relating to this original issue. 

Sorry for the misdirection @bchen1116!

Reply:
Ah, yep that makes sense! Thanks for pointing that out!